File: reload/AD/MR/Ex3.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Thu Sep 24 20:00:29 2026

@author: jn
"""

import random
import math

raio1 = 1
raio2 = 1
nSamples = 1000000
contDentroAreaCultivada = 0

for i in range(nSamples):
    x = random.random()
    y = random.random()
    
    distancia1 = math.sqrt((x-0)**2 + (y-0)**2)
    distancia2 = math.sqrt((x-1)**2 + (y-1)**2)
    
    if distancia1 <= raio1 and distancia2 <= raio2:
        contDentroAreaCultivada+=1 
        
proporcaoDentroAreaCultivada = contDentroAreaCultivada / nSamples
areaTotal = 1
areaCultivada = proporcaoDentroAreaCultivada * areaTotal
aduboPorQuilometro = 200
numeroMeses = 12
quantidadeAdubo = areaCultivada * aduboPorQuilometro * numeroMeses

print('Quantidade (kg) = ', quantidadeAdubo)


        

File: reload/AD/MR/Ex3.R

raio1 <- 1
raio2 <- 1
nSamples <- 100000
contDentroAreaCultivada <- 0

for (i in 1:nSamples)
{
  x <- runif(1)
  y <- runif(1)
  
  distancia1 <- sqrt((x-0)^2 + (y-0)^2)
  distancia2 <- sqrt((x-0)^2 + (y-0)^2)
  
  if ((distancia1 <= raio1) & (distancia2 <= raio2))
  {
    contDentroAreaCultivada <- contDentroAreaCultivada + 1      
  }
       
}

PDentroAreaCultivada <- contDentroAreaCultivada / nSamples

areaTotal <- 1

areaCultivada <- PDentroAreaCultivada * areaTotal

aduboPorQuilometro <- 200

numeroMeses <- 12

quantidadeAdubo <- areaCultivada * aduboPorQuilometro * numeroMeses

print(paste0("Quantidade (kg) = ", quantidadeAdubo))